webapp/channels/src/components/admin_console/filter/filter_checkbox.tsx

```diff
@@ -10,44 +10,42 @@ type Props = {
     updateOption: (checked: boolean, name: string) => void;
 };
 
-class FilterCheckbox extends React.PureComponent<Props> {
-    toggleOption = (e: React.MouseEvent) => {
+function FilterCheckbox(props: Props) {
+    const {name, checked, label} = props;
+
+    function toggleOption(e: React.MouseEvent) {
         e.preventDefault();
         e.stopPropagation();
-        const {checked, name, updateOption} = this.props;
+        const {checked, name, updateOption} = props;
         updateOption(!checked, name);
-    };
-
-    render() {
-        const {name, checked, label} = this.props;
-        return (
-            <div
-                className='FilterList_checkbox'
-                onClick={this.toggleOption}
-            >
-                <label>
-                    {checked &&
-                        <input
-                            type='checkbox'
-                            id={name}
-                            name={name}
-                            defaultChecked={true}
-                        />
-                    }
-
-                    {!checked &&
-                        <input
-                            type='checkbox'
-                            id={name}
-                            name={name}
-                            defaultChecked={false}
-                        />
-                    }
-                    {label}
-                </label>
-            </div>
-        );
     }
+    return (
+        <div
+            className='FilterList_checkbox'
+            onClick={toggleOption}
+        >
+            <label>
+                {checked &&
+                    <input
+                        type='checkbox'
+                        id={name}
+                        name={name}
+                        defaultChecked={true}
+                    />
+                }
+
+                {!checked &&
+                    <input
+                        type='checkbox'
+                        id={name}
+                        name={name}
+                        defaultChecked={false}
+                    />
+                }
+                {label}
+            </label>
+        </div>
+    );
 }
 
 export default FilterCheckbox;
```

Comment:
Since it is a pure component, it should use `React.memo`

Reply:
Adjusted!